src/NLog/SetupBuilderExtensions.cs

```diff
@@ -91,6 +91,14 @@ public static ISetupBuilder SetupSerialization(this ISetupBuilder setupBuilder,
         /// Loads NLog config created by the method <paramref name="configBuilder"/>
         /// </summary>
         public static ISetupBuilder LoadConfiguration(this ISetupBuilder setupBuilder, Action<ISetupLoadConfigurationBuilder> configBuilder)
+        {
+            return LoadConfiguration(setupBuilder, true, configBuilder);
+        }
+
+        /// <summary>
+        /// Loads NLog config created by the method <paramref name="configBuilder"/>
+        /// </summary>
+        public static ISetupBuilder LoadConfiguration(this ISetupBuilder setupBuilder, bool applyOnReload, Action<ISetupLoadConfigurationBuilder> configBuilder)
```

Comment:
I'm doubting if adding a bool here is a good idea. Maybe it should be an option object? Otherwise it's hard to extend. 

What do you think @snakefoot ?

Reply:
When creating the LoadConfiguration-methods here https://github.com/NLog/NLog.Web/pull/540, then I wanted them to look like `AddJsonFile`. https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.jsonconfigurationextensions.addjsonfile But the lambda makes it a little different. If you believe the number of options for the same method will be more than 3 then I think an options-object makes sense.